TrustKit/Pinning/ssl_pin_verifier.m

```diff
@@ -48,47 +49,49 @@ TSKTrustEvaluationResult verifyPublicKeyPin(SecTrustRef serverTrust, NSString *s
         return TSKTrustEvaluationErrorInvalidParameters;
     }
     
+    CFIndex certificateChainLen = SecTrustGetCertificateCount(serverTrust);
+    
     if ((trustResult != kSecTrustResultUnspecified) && (trustResult != kSecTrustResultProceed))
     {
-        // Default SSL validation failed
-        CFDictionaryRef evaluationDetails = SecTrustCopyResult(serverTrust);
-        TSKLog(@"Error: default SSL validation failed for %@: %@", serverHostname, evaluationDetails);
-        CFRelease(evaluationDetails);
-        CFRelease(serverTrust);
-        return TSKTrustEvaluationFailedInvalidCertificateChain;
+        // If no fallback pins were provided, abort immediately due to failed trust chain
+        if (fallbackKnownPins.count == 0 && certificateChainLen > 0)
+        {
+            // Default SSL validation failed
+            CFDictionaryRef evaluationDetails = SecTrustCopyResult(serverTrust);
+            TSKLog(@"Error: default SSL validation failed for %@: %@", serverHostname, evaluationDetails);
+            CFRelease(evaluationDetails);
+            CFRelease(serverTrust);
+            return TSKTrustEvaluationFailedInvalidCertificateChain;
+        }
+        else
+        {
+            // If SSL trust validation has failed, allow only emergency pin validation
+            // against the SSL leaf certificate. If this operation is successful, then
+            // we can trust the certificate, otherwise continue treating as invalid
+            // trust chain.
+            CFIndex leafCertIndex = certificateChainLen - 1;
```

Comment:
This appears to assume the "break glass" certificate is used directly. What if it is used as a CA to endorse a chain of trust. Shouldn't we loop down the chain like we do below?